backend/src/backend/filterlog.clj

```diff
@@ -15,58 +15,74 @@
 
 (ns backend.filterlog
   (:require [backend.database :refer [db]]
-            [clojure.java.jdbc :as j]))
+            [clojure.java.jdbc :as j]
+            [backend.logging :refer [get-adaption-id]]))
 
-(defn get-filtered-syslog
-  "Retrieve and filters data from the database table 'system_log'
-  based on input."
-  [{:keys [device_id adaption_id adaption_type description date date_from date_to]}]
-  (def querystr "SELECT system_log.system_log_id, system_log.device_id, system_log.adaption_id, adaption.adaption_type, adaption.adaption_description, adaption.config_id, system_log.description, system_log.created
-                FROM system_log
-                INNER JOIN adaption
-                ON system_log.adaption_id=adaption.adaption_id
-                WHERE")
-  (def firstParam true)
-
-  (if-not (= device_id nil)
-    (do (if (true? firstParam)
-          (do (def newstr (str " device_id IN (" device_id ")"))
-              (def firstParam false))
-          (def newstr (str " AND device_id IN (" device_id ")")))
-        (def querystr (str querystr newstr))))
+(defn adaption-filter
+  [adaption_type queryStr]
+  (if-not (= nil adaption_type)
+    (let [basicQueryString "SELECT system_log.*, adaption.adaption_type, adaption.adaption_id
+  FROM system_log INNER JOIN adaption ON adaption.adaption_id = system_log.adaption_id"]
+      (if-not (= queryStr basicQueryString)
+        (str queryStr " AND LOWER(adaption.adaption_type = LOWER('" adaption_type "')")
+        (str queryStr " WHERE LOWER(adaption.adaption_type) = LOWER('" adaption_type "')")))
+    queryStr))
 
-  (if-not (= adaption_id nil)
-    (do (if (true? firstParam)
-          (do (def newstr (str " system_log.adaption_id IN (" adaption_id ")"))
-              (def firstParam false))
-          (def newstr (str " AND system_log.adaption_id IN (" adaption_id ")")))
-        (def querystr (str querystr newstr))))
+(defn device-filter
+  [device_id queryStr]
+  (if-not (= nil device_id)
+    (let [basicQueryString "SELECT system_log.*, adaption.adaption_type, adaption.adaption_id
+  FROM system_log INNER JOIN adaption ON adaption.adaption_id = system_log.adaption_id"]
```

Comment:
this let binding formatting looks weird, remember to run `lein cljfmt fix` or even better add it as a pre-commit hook

Reply:
i did and tried running it again, still didn't fix these bindings.. fixed manually